charms/osm/sshproxy.py

```diff
@@ -376,20 +379,23 @@ def run(self, cmd: str) -> (str, str):
 
         # Make sure we have everything we need to connect
         if host and user:
-            return self.ssh(cmd)
+            return self.ssh(cmd, ssh_options)
 
         raise Exception("Invalid SSH credentials.")
 
-    def scp(self, source_file, destination_file):
+    def scp(self, source_file, destination_file, ssh_options: List[str] = None):
         """Execute an scp command. Requires a fully qualified source and
         destination.
 
         :param str source_file: Path to the source file
         :param str destination_file: Path to the destination file
+        :param list(str) ssh_options: The options to pass to the SCP command
         :raises: :class:`CalledProcessError` if the command fails
         """
         if which("sshpass") is None:
             SSHProxy.install()
+        if ssh_options is None:
```

Comment:
In the function argument? It is because the `ssh_options` is a mutable, in this case a list. Therefore, using `ssh_options=[]` in the function's arguments can have some unpredictable results, and even the [Python documentation](https://docs.python.org/3/reference/compound_stmts.html#function-definitions) indicates not to use this form.

An example of what might happen is shown in the [Using a mutable default value as an argument](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) article.

Reply:
In the function argument? It is because the `ssh_options` is a mutable, in this case a list. Therefore, using `ssh_options=[]` in the function's arguments can have some unpredictable results, and even the [Python documentation](https://docs.python.org/3/reference/compound_stmts.html#function-definitions) indicates not to use this form.

An example of what might happen is shown in the [Using a mutable default value as an argument](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) article.

Reply:
> if the only purpose of setting the default value of ssh_options as None is to check it later and transform it to [], why don't you set [] as the default value?

In the function argument? It is because the `ssh_options` is a mutable, in this case a list. Therefore, using `ssh_options=[]` in the function's arguments can have some unpredictable results, and even the [Python documentation](https://docs.python.org/3/reference/compound_stmts.html#function-definitions) indicates not to use this form.

An example of what might happen is shown in the [Using a mutable default value as an argument](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) article.

Reply:
ok, that makes sense thanks!!